packages/eslint-plugin/src/rules/return-await.ts

```diff
@@ -73,6 +73,53 @@ export default createRule({
       scopeInfoStack.pop();
     }
 
+    function affectsExplicitResourceManagement(node: TSESTree.Node): boolean {
+      // just need to determine if there is a `using` declaration in scope.
+      let scope = context.sourceCode.getScope(node);
+      const functionScope = scope.variableScope;
+      while (true) {
+        for (const variable of scope.variables) {
+          if (variable.defs.length !== 1) {
+            // This can't be the case for `using` or `await using` since it's
+            // an error to redeclare those more than once in the same scope,
+            // unlike, say, `var` declarations.
+            continue;
+          }
+
+          const declaration = variable.defs[0];
+          const declaratorNode = declaration.node;
+          const declarationNode =
+            declaratorNode.parent as TSESTree.VariableDeclaration;
+
+          // if it's a using/await using declaration, and it comes _before_ the
+          // node we're checking, it affects control flow for that node.
+          if (
+            ['using', 'await using'].includes(declarationNode.kind) &&
+            declaratorNode.range[1] < node.range[0]
+          ) {
+            return true;
+          }
+        }
+
+        if (scope === functionScope) {
+          // We've checked all the relevant scopes
+          break;
+        }
+
+        if (scope.upper == null) {
+          // we've hit the top level of the program.
+          // This shouldn't happen, since the rule should only be checking
+          // contexts in which `return` statements are legal, which should always
+          // be inside a function.
+          break;
+        }
+
+        scope = scope.upper;
```

Comment:
[Style] Since no tests fail with the `break;` removed:

```suggestion
        // This should always exist, since the rule should only be checking
        // contexts in which `return` statements are legal, which should always
        // be inside a function.
        scope = scope.upper!;
```

Reply:
Added a few tests to ensure that it doesn't crash or anything with this change 👍